### PR TITLE
feat: part-3 refresh session:  user session manager impl

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Implementing user session manager

### Dependencies (Optional)
#302 
#304 

----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by typfel** [comment]
@MohamadJaara will you rebase this PR?

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt [hunk 1]
```diff
@@ -27,34 +26,30 @@ import kotlinx.coroutines.runBlocking
...
-
+            val networkContainer = AuthenticatedNetworkContainer(SessionManagerImpl(sessionRepository, session.userId, sessionMapper, serverConfigMapper))
             val proteusClient: ProteusClient = ProteusClientImpl(rootProteusDirectoryPath, session.userId)
             runBlocking { proteusClient.open() }
 
             val workScheduler = WorkScheduler(appContext, session)
             val syncManager = SyncManagerImpl(workScheduler)
             val encryptedSettingsHolder = EncryptedSettingsHolder(appContext, "${PREFERENCE_FILE_PREFIX}-${session.userId}")
             val userPreferencesSettings = KaliumPreferencesSettings(encryptedSettingsHolder.encryptedSettings)
+            // FIXME: should the DB name be unique per user ?
```
Comment (typfel) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt:52]: Yes

**Review by typfel — COMMENTED** [review]
I guess the interface of `SessionManager` will need to change when you implement the TODOs.

Any reason for not implementing that now in this PR?

**Comment by MohamadJaara** [comment]
There are 3 To-dos in `SessionManagerImpl` were left intensionally for the issue of session don't exist/expired that need to be handled separately  in a different PR

**Review by typfel — APPROVED** [review]

**Comment by MohamadJaara** [comment]
The only change left for part-4, is to change `WorkScheduler` and `UserSessionScope` to depend on a User ID instead of `AuthSession`

**Pull request merged**